res/controllers/Pioneer-DDJ-200-scripts.js

```diff
@@ -1,41 +1,133 @@
 var DDJ200 = {
-    jogDisabled: new Array(false, false, false, false, false)
+    fourDeckMode: false,
+    vDeckNo: [0, 1, 2],
+    vDeck: {},
+    shiftPressed: {left: false, right: false},
+    jogCounter: 0
 };
 
-DDJ200.init = function() {};
+DDJ200.init = function() {
+    for (var i = 1; i <= 4; i++) {
+                                  
+       // create associative arrays for 4 virtual decks
+        this.vDeck[i] = {
+        syncEnabled: false,
+        volMSB: 0,
+        rateMSB: 0,
+        jogDisabled: false,
+        };
 
-DDJ200.shutdown = function() {};
+        var vgroup = "[Channel" + i + "]";
 
-DDJ200.scratch = function(channel, control, value, status, group) {
+        // run onTrackLoad after every track load to set LEDs accordingly
+        engine.makeConnection(vgroup, "track_loaded", function(ch, vgroup) {
+                DDJ200.onTrackLoad(ch, vgroup);
+            });
+
+        // set Pioneer CDJ cue mode for all decks
+         engine.setValue(vgroup, "cue_cdj", true);
+    }
+
+    DDJ200.LEDsOff();
+
+    // start with focus on library for selecting tracks (delay seems required)
+    engine.beginTimer(500, function() {
+            engine.setValue("[Library]", "MoveFocus", 1);
+        }, true);
+};
+
+DDJ200.shutdown = function() { DDJ200.LEDsOff(); };
+
+DDJ200.LEDsOff = function() {                         // trun off LED buttons:
+
+    for (var i = 0; i <= 1; i++) {
+        midi.sendShortMsg(0x96 + i, 0x63, 0x00);      // set headphone master
+        midi.sendShortMsg(0x90 + i, 0x54, 0x00);      // pfl headphone
+        midi.sendShortMsg(0x90 + i, 0x58, 0x00);      // beat sync
+        midi.sendShortMsg(0x90 + i, 0x0B, 0x00);      // play
+        midi.sendShortMsg(0x90 + i, 0x0C, 0x00);      // cue
+        for (var j = 0; j <= 8; j++) {
+            midi.sendShortMsg(0x97 + 2 * i, j, 0x00); // hotcue
+        }
+    }
+};
+
+DDJ200.onTrackLoad = function(channel, vgroup) {
+    // set LEDs (hotcues, etc.) for the loaded deck
+    // if controller is switched to this deck
+    var vDeckNo = script.deckFromGroup(vgroup);
+    var deckNo = 2; if (vDeckNo % 2) deckNo = 1;
```

Comment:
missing braces and linebreak.

why not this?
```suggestion
    var deckNo = (vDeckNo % 2) ? 1 : 2;
```